### DISCUSSION TITLE
How would I use this to validate icu strings in an .arb file?

### DISCUSSION THREAD

**Discussion by robbecker-wf** [unicode-org/icu4x#3163] [Q&A] [answered]
For context, I'd like to use  icu4x to parse .arb files to validate the syntax of the strings before sending them for translation.
An .arb file is primarily a Dart/Flutter file format for string translations. So the solution doesn't have to be run from Dart if there's something that will already validate an .arb file written in Rust and I can just run the binary. Any thoughts or help someone could provide here? Do I just need to wait for Dart support and write my own validator?

**Comment by Manishearth** [comment, marked as answer]
I do not think this is the right library for that task; you probably should find a `.arb` parser that does it. ICU4X does plan to support MessageFormat 2.0 when it's ready, and ICU4X fully intends to provide all of the internationalization components needed by libraries that handle translation strings, but not actually the parsing of other formats.
